### ISSUE TITLE
OPO: Bluetooth MAC address is invalid

### ISSUE THREAD

**Issue by Flohack74** [ubports/ubuntu-touch#955]
This is related to https://github.com/ubports/ubuntu-touch/issues/541 - Also like the WiFi MAC address the Bluetooth address seems to be the same for all users, and it seems to be an invalid one, starting with a lot of zeroes.

**Comment by ntorquet** [comment]
Hi!
I confirm! In my OPO that I just install, the wifi mac address is B0:11:33:88:88:88 and bluetooth 00:00:00:00:5A:AD

**Comment by chriscroome** [comment]
I have the Bluetooth address `00:00:00:00:5A:AD` and I just connected to a Bluetooth speaker without a problem, however I couldn't connect to a Bluetooth mouse.

**Comment by Vince1171** [comment]
same on Oneplus 3/3T (port in progress), bluetooth adress is ```00:00:00:00:5A:AD```

**Comment by Flohack74** [comment]
We are missing something that is setting both WiFi and BT MAC correctly in Android. @Vince1171 can we go over initrc scripts from Android one day, to see that maybe we miss a trigger or so? The problem is also that my "fix" for WiFi broke it actually for numerous people, since there are 2 versions of OPO: bacon and A0001, and they handle the /persist partition differently - on bacon it seems to be empty. 
Rather than hacking into persist partition the container should load the MACs from NVRAM, for which there is a special service probably.

**Comment by Vince1171** [comment]
Sure we can. 
Oh, I didn't there was 2 type of OPO.
But I have to warn you, bluetooth on op3 is super unstable, it works only once in five.
Bluetooth is a nightmare ^^

**Comment by wdehoog** [comment]
@Flohack74 @Vince1171  while investigating this issue I noticed hci_qcomm_init (in init_wlan_bt.sh) has some problems. 
I am now using this and hciconfig reports a 'normal' bt mac address (same as the wifi one):

```
        echo 0 > /sys/module/hci_smd/parameters/hcismd_set
        /system/bin/logwrapper -k /system/bin/hci_qcomm_init -vvv -e
        sleep 1 
        echo 1 > /sys/module/hci_smd/parameters/hcismd_set
```

Some remarks:

I don't know why hcismd_set must be turnd off first but it happens in other distributions and if not turned off hci_qcomm_init complains about /dev/smd3
```
init_transport: Cannot open /dev/smd3: No such device
```

Enabling logging for hci_qcomm_init seems like a good plan to me.

The `sleep 1` seems useless.

The line `logwrapper /system/bin/conn_init` should be removed as there is no conn_init in /system/bin.

**Comment by Flohack74** [comment]
Ok I will try this....

**Comment by mariogrip** [comment]
Fixed pushed, so closes for QA

**Comment by Flohack74** [comment]
Oh dang, I missed that automation did not close xD

**Comment by Flohack74** [comment]
@wdehoog Can you please check if you see the same, now BT and WiFi MAC address are identical. I am not sure this is correct. It will resolve however the primary problem as now at least every OPO has a unique BT address.
Can I talk to you on Telegram, I have another issue around that topic.
